cmd/finch/version_test.go

```diff
@@ -32,17 +37,68 @@ func captureStdout(t *testing.T, f func()) string {
 func TestVersionCommand(t *testing.T) {
 	t.Parallel()
 
-	cmd := newVersionCommand()
+	ctrl := gomock.NewController(t)
+	logger := mocks.NewLogger(ctrl)
+	lcc := mocks.NewLimaCmdCreator(ctrl)
+	cmd := newVersionCommand(lcc, logger)
 	assert.Equal(t, cmd.Name(), "version")
 }
 
-//nolint:paralleltest // TODO: Add t.Parallel after using dependency injection to pass a io.Writer to newVersionCommand.
-func TestVersionAction(t *testing.T) {
-	mockCmd := newVersionCommand()
-	versionActionFunc := func() {
-		require.NoError(t, versionAction(mockCmd, []string{}))
+func TestPrintVersion(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name    string
+		command *cobra.Command
+		args    []string
+		mockSvc func(*mocks.LimaCmdCreator, *mocks.Logger, *gomock.Controller)
+	}{
+		{
+			name: "print finch version",
+			command: &cobra.Command{
+				Use: "version",
+			},
+			args: []string{},
+			mockSvc: func(lcc *mocks.LimaCmdCreator, logger *mocks.Logger, ctrl *gomock.Controller) {
+				getVMStatusC := mocks.NewCommand(ctrl)
+				lcc.EXPECT().CreateWithoutStdio("ls", "-f", "{{.Status}}", limaInstanceName).Return(getVMStatusC)
+				getVMStatusC.EXPECT().Output().Return([]byte("Running"), nil)
+				logger.EXPECT().Debugf("Status of virtual machine: %s", "Running")
+			},
+		},
 	}
 
-	output := captureStdout(t, versionActionFunc)
-	assert.Contains(t, output, finchVersion)
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			ctrl := gomock.NewController(t)
+			logger := mocks.NewLogger(ctrl)
+			lcc := mocks.NewLimaCmdCreator(ctrl)
+
+			versionActionFunc := func() {
+				require.NoError(t, newVersionAction(lcc, logger).printVersion())
+			}
+		
+			output := captureStdout(t, versionActionFunc)
```

Comment:
https://github.com/runfinch/finch/pull/83#discussion_r1065119672 is also applicable to this PR.

Reply:
Updated on the latest push. Can you check and suggest how to assert through `&stdOut'

Reply:
I’ve been trying to assert through this,
```var output bytes.Buffer
			newVersionAction(lcc, logger, &output).run()
			assert.Contains(t, output.String(), tc.versionOutput.Client)```
But during debug, i can see that `&output`  buffer is not getting filled through the `io.Writer`